superset/tables/models.py

```diff
@@ -0,0 +1,92 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Table model.
+
+This model was introduced in SIP-68 (https://github.com/apache/superset/issues/14909),
+and represents a "table" in a given database -- either a physical table or a view. In
+addition to a table, new models for columns, metrics, and datasets were also introduced.
+
+These models are not fully implemented, and shouldn't be used yet.
+"""
+
+from typing import List
+
+import sqlalchemy as sa
+from flask_appbuilder import Model
+from sqlalchemy.orm import backref, relationship
+from sqlalchemy.schema import UniqueConstraint
+
+from superset.columns.models import Column
+from superset.models.core import Database
+from superset.models.helpers import (
+    AuditMixinNullable,
+    ExtraJSONMixin,
+    ImportExportMixin,
+)
+
+association_table = sa.Table(
+    "sl_table_columns",
+    Model.metadata,  # pylint: disable=no-member
+    sa.Column("table_id", sa.ForeignKey("sl_tables.id")),
+    sa.Column("column_id", sa.ForeignKey("sl_columns.id")),
+)
+
+
+class Table(Model, AuditMixinNullable, ExtraJSONMixin, ImportExportMixin):
+    """
+    A table/view in a database.
+    """
+
+    __tablename__ = "sl_tables"
+
+    # Note this uniqueness constraint is not part of the physical schema, i.e., it does
+    # not exist in the migrations. The reason it does not physically exist is MySQL,
+    # PostgreSQL, etc. have a different interpretation of uniqueness when it comes to NULL
+    # which is problematic given the catalog and schema are optional.
+    __table_args__ = (UniqueConstraint("database_id", "catalog", "schema", "name"),)
```

Comment:
@betodealmeida It's also not possible to apply this constraint in MySQL because TEXT cannot be used in index keys without specifying a length: https://stackoverflow.com/questions/1827063/mysql-error-key-specification-without-a-key-length